main.tf

```
@@ -9,6 +9,9 @@ locals {
 
   final_namespace = var.create_namespace ? resource.kubernetes_namespace_v1.this[0].metadata[0].name : data.kubernetes_namespace_v1.this[0].metadata[0].name
 
+  use_existing_project = var.gitlab_project_name == "" ? 1 : 0
+  project_id           = local.use_existing_project == 1 ? data.gitlab_project.this[0].id : gitlab_project.project[0].id
+
```

```suggestion
  project_id                                      = local.use_existing_project == 1 ? data.gitlab_project.this[0].id : gitlab_project.project[0].id
  project_path_with_namespace = local.use_existing_project == 1 ? data.gitlab_project.this[0].path_with_namespace : gitlab_project.project[0].path_with_namespace

```

I think we can manage conditional variable here in order to keep the code below easier to read. What do you think?